cmd/broker/ingress/main.go

```diff
@@ -213,11 +251,12 @@ func (r *runnableServer) Start(stopCh <-chan struct{}) error {
 			ctx = context.Background()
 		}
 		logger.Info("Shutting down...")
-		if err := r.s.Shutdown(ctx); err != nil {
+		if err = r.s.Shutdown(ctx); err != nil {
 			logger.Error("Shutdown returned an error", zap.Error(err))
 		} else {
 			logger.Info("Shutdown done")
 		}
-	}()
-	return r.s.ListenAndServe()
+	}
+	r.wg.Done()
```

Comment:
I _think_ this might panic if for some reason the stopCh was already closed. As the starting go routine might not have had time to start, so calling Done() may cause it to go below zero, and thus panic.

Reply:
Good point. Maybe this should go inside the serving goroutine? I think that would have the same behavior but without the possible panic.